### PR TITLE
Improve WordPress plugin error handling and messages, set version to 0.0.4

### PR DESCRIPTION
**Pull request by CiaranMn**
## 🌟 What is the purpose of this PR?

Improved WordPress plugin error handling and bug fixes for the `0.0.4` release:
1. Add `Toast` component for better user-facing errors than default WordPress admin notices / snackbar: deduplicated and visible wherever user is scrolled to in page
2. Show toast when entities cannot be created or retrieved (rather than error being in browser console only)
3. Fix a bug in 'find entity location' logic by retrieving all pages to look in (rather than the default 5 pages)
4. Don't show block in selector if DB version is unsupported
5. Add label to custom block controls in settings sidebar
6. Set a shorter db connection timeout in Block Protocol API (we are seeing intermittent errors in connecting to the db, and the default timeout is 30s)
7. Reduce timeout in plugin for connecting to the BP API from 10s to 5s (if 5s is reached the API is definitely not going to respond successfully)

Also sets the necessary metadata for the `0.0.4` release, and documents the publishing process.

## 🔗 Related links

- [Asana task for better error toasts](https://app.asana.com/0/1203358502199087/1204145931939225/f) _(internal)_
- [Asana task for custom block setting labelling](https://app.asana.com/0/1203358502199087/1204182825551978/f) _(internal)_

## ⚠️ Known issues

- Intermittent timeouts in connection to MongoDB – no luck in finding cause so far (sometimes none a day, sometimes a few).

## ❓ How to test this?

1.  Failed entity retrieval can be tested by adding a block to a page and then deleting the entity it is associated with in the db and refreshing
2. External service calls can be tested by hitting a usage limit (easiest to do by retrieving static maps repeatedly) and then trying another request
3. Block settings panel can be checked by selecting a BP block and looking at righthand settings sidebar

## 📹 Demo

https://user-images.githubusercontent.com/37743469/225738021-c5603866-063e-43f7-ba41-31203b5199df.mp4

### CODE REVIEW

**Comment by CiaranMn** [comment]
> @CiaranMn Can we make the toast less wide?

What proportion of a large screen should it be?

File: libs/wordpress-plugin/plugin/trunk/block/edit-or-preview/edit/service-callbacks.ts [hunk 1]
```diff
@@ -33,7 +39,7 @@ export const callService = async ({
   });
 
   if ("error" in apiResponse) {
-    let errorMessage = apiResponse.error ?? "An unknown error occured";
+    let errorMessage = apiResponse.error ?? "An unknown error occurred";
```
Comment (benwerner01) [inline libs/wordpress-plugin/plugin/trunk/block/edit-or-preview/edit/service-callbacks.ts:42]: Apologies if I'm missing something obvious, but why would the error message ever be `apiResponse.error`? We're nesting the error message in an `errors` array in the BP API if I recall correctly.

**Review by benwerner01 — DISMISSED** [review]
Thanks Ciaran, the changes look good to me and the toast is working as expected for me locally.

File: libs/wordpress-plugin/plugin/trunk/block/edit-or-preview/edit/service-callbacks.ts (hunk 1 above)
Comment (CiaranMn) [reply libs/wordpress-plugin/plugin/trunk/block/edit-or-preview/edit/service-callbacks.ts:42]: This the WordPress API, where we only extract a single error (or at least we do in addition to providing the original BP response) – see [here](https://github.com/blockprotocol/blockprotocol/blob/f4aad130c59e3d2c8bdcf154322bd1b1dd9214fa/libs/wordpress-plugin/plugin/trunk/server/block-api-endpoints.php#LL649)

File: libs/wordpress-plugin/plugin/trunk/block/edit-or-preview/edit.tsx [hunk 2]
```diff
@@ -52,6 +54,19 @@ export const Edit = ({
   const [entitySubgraph, setEntitySubgraph] =
     useState<Subgraph<EntityRootType> | null>(null);
 
+  const displayToast = useCallback(
+    (toastProps: ToastProps, options?: Parameters<typeof toast>[1]) => {
+      toast(<Toast {...toastProps} />, {
```
Comment (thehabbos007) [inline libs/wordpress-plugin/plugin/trunk/block/edit-or-preview/edit.tsx:59]: The action text is green, is this intended? It looks slightly out of place
image

File: libs/wordpress-plugin/plugin/trunk/block/edit-or-preview/edit.tsx [hunk 3]
```diff
@@ -109,14 +141,26 @@ export const Edit = ({
           entityId,
           graphResolveDepths: blockSubgraphResolveDepths,
         },
-      }).then(({ data }) => {
+      }).then(async ({ data, errors }) => {
```
Comment (thehabbos007) [inline libs/wordpress-plugin/plugin/trunk/block/edit-or-preview/edit.tsx:144]: Does this need to be `async`?

File: libs/wordpress-plugin/plugin/trunk/block/edit-or-preview/edit.tsx [hunk 4]
```diff
@@ -130,20 +174,34 @@ export const Edit = ({
...
     setEntitySubgraph(subgraph);
-  }, [entityId]);
+  }, [displayToast, entityId]);
 
   const serviceCallbacks = useMemo<ServiceEmbedderMessageCallbacks>(
-    () => constructServiceModuleCallbacks(),
-    [],
+    () =>
+      constructServiceModuleCallbacks((toastProps) =>
+        displayToast(toastProps, { toastId: "billing" }),
```
Comment (thehabbos007) [inline libs/wordpress-plugin/plugin/trunk/block/edit-or-preview/edit.tsx:202]: This seems to be the only place where we explicitly specify a `toastId`. Does this mean we don't stack the errors for equivalent toasts?

**Review by thehabbos007 — COMMENTED** [review]
Thanks, Ciaran! I have some comments here. I'll also aim to open a PR to add Sentry JS to admin editing before the release goes out

File: libs/wordpress-plugin/plugin/trunk/block/edit-or-preview/edit.tsx (hunk 3 above)
Comment (CiaranMn) [reply libs/wordpress-plugin/plugin/trunk/block/edit-or-preview/edit.tsx]: Nope! removed, thank you

File: libs/wordpress-plugin/plugin/trunk/block/edit-or-preview/edit.tsx (hunk 4 above)
Comment (CiaranMn) [reply libs/wordpress-plugin/plugin/trunk/block/edit-or-preview/edit.tsx:224]: This is the only error that will be repeated in short succession – the others won't be so we don't need to deduplicate them

File: libs/wordpress-plugin/plugin/trunk/block/edit-or-preview/edit.tsx (hunk 2 above)
Comment (CiaranMn) [reply libs/wordpress-plugin/plugin/trunk/block/edit-or-preview/edit.tsx]: I have updated the toast styling in https://github.com/blockprotocol/blockprotocol/pull/1181/commits/081b445281c6c7138ef94d708668db11fd325b31 – please see new demo in description

**Review by thehabbos007 — APPROVED** [review]
Looks good!

**Review by thehabbos007 — DISMISSED** [review]

File: libs/wordpress-plugin/plugin/trunk/readme.txt [hunk 5]
```diff
@@ -5,23 +5,24 @@ Tags: block protocol, blocks, gutenberg, gutenberg blocks, block, schema, countd
...
 
-Discover and use new blocks when you need them, directly from within the WordPress Gutenberg Editor. No more installing new plugins to get more blocks.
+Discover and use new blocks when you need them, at the point of insertion, directly from within the WordPress editor. No more installing new plugins to get more blocks.
 
-In addition to useful blocks like Callouts, Headings, Images, and Quotes, the Block Protocol plugin provides powerful, new blocks for services such as OpenAI, and Mapbox.
+- Blocks that provide access to external services including **OpenAI** and **Mapbox**, without needing to sign up for these separately
+- Structured data blocks with SEO benefits for your website, helping you rank more highly in Google and Bing (e.g. the [Address](https://blockprotocol.org/@hash/blocks/address) and [How-to](https://blockprotocol.org/@hash/blocks/how-to) blocks)
+- Interactive blocks such as the [drawing](https://blockprotocol.org/@tldraw/blocks/drawing) and [countdown timer](https://blockprotocol.org/@hash/blocks/timer) blocks
+- "Traditional" content management blocks such as [callouts](https://blockprotocol.org/@hash/blocks/callout), headings, images, and quotes
```
Comment (nonparibus) [inline libs/wordpress-plugin/plugin/trunk/readme.txt:23]:
```suggestion
- "Traditional" content management blocks such as [callouts](https://blockprotocol.org/@hash/blocks/callout), [tables](https://blockprotocol.org/@hash/blocks/table), headings, images, and quotes
```

With the v2 (and now fixed) table block providing a (sometimes better) alternative to the WP native one, depending on what you're trying to do...

**Pull request merged**